source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -0,0 +1,194 @@
+#include "common/tracing/zipkin/zipkin_tracer_impl.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/http/headers.h"
+#include "common/http/message_impl.h"
+#include "common/http/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+#include "common/tracing/zipkin/zipkin_core_constants.h"
+
+#include "spdlog/spdlog.h"
+
+namespace Zipkin {
+
+ZipkinSpan::ZipkinSpan(Zipkin::Span& span) : span_(span) {}
+
+void ZipkinSpan::finishSpan() { span_.finish(); }
+
+void ZipkinSpan::setTag(const std::string& name, const std::string& value) {
+  if (this->hasCSAnnotation()) {
+    span_.setTag(name, value);
+  }
+}
+
+bool ZipkinSpan::hasCSAnnotation() {
+  auto annotations = span_.annotations();
+  if (annotations.size() > 0) {
+    // We currently expect only one annotation to be in the span when this function is called.
+    return annotations[0].value() == ZipkinCoreConstants::get().CLIENT_SEND;
+  }
+  return false;
+}
+
+ZipkinDriver::TlsZipkinTracer::TlsZipkinTracer(TracerPtr tracer, ZipkinDriver& driver)
```

Comment:
nit: for clarify making this take ```TracerPtr&&``` which will make it clear at the call site that you are moving.